dotnet/src/SemanticKernel/Text/TextChunker.cs

```diff
@@ -131,31 +131,34 @@ private static List<string> InternalSplitTextParagraphs(List<string> lines, int
 
     private static List<string> BuildParagraph(List<string> truncatedLines, StringBuilder paragraphBuilder, List<string> paragraphs, int maxTokensPerParagraph, Func<string, int, List<string>> longLinesSplitter)
     {
-        // Base case: no more elements in the list
-        if (truncatedLines.Count == 0)
+        while (true)
```

Comment:
nit: Looks like the only condition to escape from infinite loop is `truncatedLines.Count == 0`, so probably we can replace `while (true)` with `while (truncatedLines.Count > 0)`?

Unit-tests are passed with this approach:
```csharp
while (truncatedLines.Count > 0)
{
    string line = truncatedLines[0];

    if (paragraphBuilder.Length > 0 && TokenCount(paragraphBuilder.Length) + TokenCount(line.Length) + 1 >= maxTokensPerParagraph)
    {
        paragraphs.Add(paragraphBuilder.ToString().Trim());

        // next paragraph
        paragraphBuilder.Clear();
    }
    else
    {
        paragraphBuilder.AppendLine(line);

        truncatedLines.RemoveAt(0);
    }
}

// Adding any remaining paragraph
if (paragraphBuilder.Length > 0)
{
    paragraphs.Add(paragraphBuilder.ToString().Trim());
}

return paragraphs;
```

Reply:
Yes, it's the same thing, just reorganized.  I was trying to minimize churn, but I can restructure it if that's desired.